crates/uv/tests/pip_install_scenarios.rs

```diff
@@ -2368,14 +2449,56 @@ fn post_local_greater_than_post() {
     Resolved 1 package in [TIME]
     Downloaded 1 package in [TIME]
     Installed 1 package in [TIME]
-     + package-a==1.2.3.post1
+     + package-a==1.2.3.post0
     "###);
 
-    // The version '1.2.3.post1' satisfies the constraint '>1.2.3.post0'.
+    // The version '1.2.3.post0' satisfies the constraint '==1.2.3.post0'.
     assert_installed(
         &context.venv,
-        "post_local_greater_than_post_a",
-        "1.2.3.post1",
+        "post_equal_available_a",
+        "1.2.3.post0",
+        &context.temp_dir,
+    );
+}
+
+/// A greater-than version constraint should not match a post-release version if the
+/// post-release version is not available.
+///
+/// ```text
+/// post-greater-than-post-not-available
+/// ├── environment
+/// │   └── python3.8
+/// ├── root
+/// │   └── requires a>1.2.3.post2
+/// │       └── unsatisfied: no matching version
+/// └── a
+///     ├── a-1.2.3
+///     ├── a-1.2.3.post0
+///     └── a-1.2.3.post1
+/// ```
+#[test]
+fn post_greater_than_post_not_available() {
+    let context = TestContext::new("3.8");
+
+    // In addition to the standard filters, swap out package names for shorter messages
+    let mut filters = INSTA_FILTERS.to_vec();
+    filters.push((r"post-greater-than-post-not-available-", "package-"));
+
+    uv_snapshot!(filters, command(&context)
+        .arg("post-greater-than-post-not-available-a>1.2.3.post2")
+        , @r###"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+
+    ----- stderr -----
+      × No solution found when resolving dependencies:
+      ╰─▶ Because only package-a<1.2.3.post3 is available and you require package-a>=1.2.3.post3, we can conclude that the requirements are unsatisfiable.
```

Comment:
This is not totally ideal (though it is _correct_) because the user asked for `>1.2.3.post2`. I think it's fine though. (These are rare anyway.)

Reply:
Yeah, when we see `post-greater-than-post-not-available-a>1.2.3.post2`, we internally rewrite it as `post-greater-than-post-not-available-a>=1.2.3.post3`, since that's the easiest way to represent the constraint.